src/DIRAC/WorkloadManagementSystem/Service/JobMonitoringHandler.py

```diff
@@ -56,27 +56,27 @@ def getJobsAttributes(cls, *args, **kwargs):
     types_getApplicationStates = []
 
     @classmethod
-    def export_getApplicationStates(cls, condDict=None, older=None, newer=None):
+    def export_getApplicationStates(cls):
         """Return Distinct Values of ApplicationStatus job Attribute in WMS"""
-        return cls.jobDB.getDistinctJobAttributes("ApplicationStatus", condDict, older, newer)
+        return cls.jobDB._query("SELECT DISTINCT ApplicationStatus FROM JobsHistorySummary")
 
     ##############################################################################
     types_getJobTypes = []
 
     @classmethod
-    def export_getJobTypes(cls, condDict=None, older=None, newer=None):
+    def export_getJobTypes(cls):
         """Return Distinct Values of JobType job Attribute in WMS"""
-        return cls.jobDB.getDistinctJobAttributes("JobType", condDict, older, newer)
+        return cls.jobDB._query("SELECT DISTINCT JobType FROM JobsHistorySummary")
 
     ##############################################################################
     types_getOwners = []
 
     @classmethod
-    def export_getOwners(cls, condDict=None, older=None, newer=None):
+    def export_getOwners(cls):
         """
         Return Distinct Values of Owner job Attribute in WMS
         """
-        return cls.jobDB.getDistinctJobAttributes("Owner", condDict, older, newer)
+        return cls.jobDB._query("SELECT DISTINCT Owner FROM JobsHistorySummary")
```

Comment:
I am not sure that there is any benefit in these changes, given that the Job table has an index on all these parameters. I just tried on the LHCb production instance, it returns the answer in 0.021s for Owner for example

Reply:
It will probably be even less for the summary table. I agree with you that it's minor improvement anyway.